### PR TITLE
[RFR] Review AppBar on desktop 

### PR DESCRIPTION
**Pull request by djhi**
- [x] Put loading indicator at the AppBar center
- [x] Put the logout button in the AppBar
- [x] Review `Responsive` to allow `null`

### Desktop

screen

### Mobile

screen

screen

### CODE REVIEW

File: UPGRADE.md [hunk 1]
```diff
@@ -1175,6 +1175,61 @@ const mapStateToProps = state => ({
 export default connect(mapStateToProps)(Menu);
 ```
 
+## Logout is now displayed in the AppBar on desktop
+
+The Logout button is now displayed in the AppBar on desktop but is still displayed as a menu item on small devices. This impact how you build a custom menu as you'll now have to check whether you are on small devices before displaying the logout:
```
Comment (fzaninotto) [inline UPGRADE.md]: s/This impact/This impacts/
s/a custom menu as/a custom menu, as/

File: packages/react-admin/src/mui/layout/Layout.js [hunk 2]
```diff
@@ -85,7 +84,7 @@ const Layout = ({
         >
             <div className={classes.appFrame}>
                 <Hidden xsDown>
-                    <AppBar title={title} open={open} />
+                    <AppBar title={title} open={open} logout={logout} />
```
Comment (fzaninotto) [inline packages/react-admin/src/mui/layout/Layout.js]: This is a BC break for people with Custom layout. To be added to the UPGRADE guide, and please update the Theming (or Admin?) section about custom layouts.

Comment (djhi) [reply packages/react-admin/src/mui/layout/Layout.js]: Added

File: UPGRADE.md [hunk 3]
```diff
@@ -1175,6 +1175,290 @@ const mapStateToProps = state => ({
...
+
+export default withRouter(connect(mapStateToProps)(Menu));
+```
+
+It also impacts custom layouts if you're using the default `AppBar` in it. You now have to pass the `logout` prop to the `AppBar`:
+
+```jsx
+// in src/MyLayout.js
+// Before
+import React, { createElement, Component } from 'react';
```
Comment (alexisjanvier) [inline UPGRADE.md]: Does all the code concern the `logout` ?  Because otherwise it's hard to see what changes affect the `logout`.

Comment (djhi) [reply UPGRADE.md]: I just committed a simplified version, is it better ?

Comment (alexisjanvier) [reply UPGRADE.md]: :+1:

File: UPGRADE.md [hunk 4]
```diff
@@ -1175,6 +1175,156 @@ const mapStateToProps = state => ({
...
+);
+
+const mapStateToProps = state => ({
+    resources: getResources(state),
+});
+
+export default withRouter(connect(mapStateToProps)(Menu));
+```
+
+It also impacts custom layouts if you're using the default `AppBar` in it. You now have to pass the `logout` prop to the `AppBar`:
```
Comment (fzaninotto) [inline UPGRADE.md]: s/ in it//

**Pull request merged**